### ISSUE TITLE
        0 : { sorter: false } option disable all the elements inside that column

### ISSUE THREAD

**Issue by gitpuneetsvyas** [Mottie/tablesorter#1510]
Hi Mottie,
suppose you have <th> which includes a drop down  and a label making  0 : { sorter: false } makes disable the entire th locked i want drop down to be keep displaying while sorting should be disabled i am using tablesorter 1.1.0 version since this is a legacy app.

my sample code

```html
<th class="sorter-false" style="width:25%" align="left"  id="byRentalStatus">
STATUS
						
	<select disabled="false" class="sorter-false" align="right"  id="rentalStatus" onChange="filterRentalAgreementByStatus('filterDataForMYRental','SearchRentalAgreementsServlet');">
						
	<c:forEach var="rentalStatus" items="${RentalAgreementsSearchBean.rentalStatusList}">
	<option value="${rentalStatus.id}" ${rentalStatus.id==filterRentalStatusId ? 'selected="selected"':""}>${rentalStatus.getDescription()}</option>
	</c:forEach>
</select>
</th>
```

**Comment by gitpuneetsvyas** [comment]
any pointers/expert comment  kindly suggest

**Comment by Mottie** [comment]
Hi @gitpuneetsvyas!

I'm not completely sure what you're asking. I've never seen tablesorter v1.1.0, the earliest version I've ever worked on was v2.0.5b. Are you sure you're using tablesorter?

**Comment by gitpuneetsvyas** [comment]
Yes sir

**Comment by Mottie** [comment]
Please share an example of the JS and HTML.

You can modify [this demo](http://jsfiddle.net/Mottie/eY8uH/25/) of the original tablesorter, or [this demo](https://jsfiddle.net/Mottie/bbxxomhx/) using my fork of tablesorter. I'll be unavailable for a few hours, so no need to rush.

**Comment by gitpuneetsvyas** [comment]
Hi Mottie,

this is the example i created  using old version of jquery and table sorter which we are using 
[https://jsfiddle.net/cpuneetvyas/o2gxgz9r/21973/](url)

i want to keep drop down enabled while disabling the sorting on that column .
i want to remove class="header headerSortUp" as per my example shown in this post however keeping drop down enabled
Please suggest any pointers

**Comment by Mottie** [comment]
That older version doesn't ignore form elements within the header. It was fixed in v2.17.1; but I would recommend updating to the latest version.

**Comment by gitpuneetsvyas** [comment]
Thanks sir is there any other way or workaround to do so without changing the version;i tried  with JQUERY/DOM way of enabling the drop down however i didn't have any good luck.
Looking forward for your suggestion.

**Comment by gitpuneetsvyas** [comment]
I have replaced my old version  jquery.tablesorter.js with new version but i was not seen drop down being enabled nor i see sorting images(asc and dsc) bein applied on the column.
Please suggest

**Comment by Mottie** [comment]
Look in the developer console (press <kbd>F12</kbd>, console tab) for any errors. Then share any error messages here, and please provide the code you're using to initialize tablesorter.

**Comment by gitpuneetsvyas** [comment]
I don't see any error on console;
this is the initialization code
`
$("table.tablesorter").jdquoteTableSorter({
				hideSearchResultsLabel: true,
				columnSorting: {
	/* Checkboxes */		0: { sorter: "checkbox" },
	/* Demo Form Number */	1: { sorter: "digit" },
							2: { sorter: false },
							/*2:{sorter: "select"} ,*/
	/* Start Date */		4: { sorter: "digit" },
	/* End Date */			5: { sorter: "jdquoteDate" }
				},
				cancelSelection :false
				
				
			
			
			});
		});
	})(jQuery); 

`

**Comment by Mottie** [comment]
I updated [this demo](https://jsfiddle.net/o2gxgz9r/22475/) to use tablesorter v2.17.1 to show that everything works... The custom code above shouldn't effect the select.

I'd still recommend using at least jQuery v1.12.4 and tablesorter v2.29.5.

**Comment by gitpuneetsvyas** [comment]
Thanks Mottie since i am using legacy app we are bound to not change the table sorter version.
we are using  table sorter 2.0.3 and jquery  1.3.1.min.js. i wrote below code remove the header,headerup and header down css classes as well as the header click event however still dodn't able to enable the dropdown.
Could you please suggest any turnaround?

**Comment by gitpuneetsvyas** [comment]
Please suggest work around to enable dropdown


`
	$(document).ready(function() {
			
			
			 $(window).load(function() { 
				 alert("page loaded");
				$('#mytable thead:nth-child(1) th:nth-child(3)').removeClass('header');
				$('#mytable thead:nth-child(1) th:nth-child(3)').removeClass('headerSortDown');
				$('#mytable thead:nth-child(1) th:nth-child(3)').removeClass('headerSortUp');
				
				$('#mytable thead:nth-child(1) th:nth-child(3)').addClass('rentalStatus');;
				$('#mytable thead:nth-child(1) th:nth-child(3)').unbind('click');
				 
				
				 alert("page removed");
				 
				 
			     //insert all your ajax callback code here. 
			     //Which will run only after page is fully loaded in background.
			  });

));


`

**Comment by gitpuneetsvyas** [comment]
Kindly suggest

**Comment by Mottie** [comment]
There's nothing I can suggest to do with options or additional code with the version of tablesorter being used, as the problem is the internal code and how it doesn't account for form elements in the header cells.

It really comes down to three choices:

1. Update tablesorter (I recommend this).
2. Modify the tablesorter code yourself to get this to work.
3. Use a different table sorting program.

**Comment by gitpuneetsvyas** [comment]
Could you please suggest which part of the table sorter plugin i will need to modify so that drop down will be enabled.
kindly give some pointers

**Comment by Mottie** [comment]
> Could you please suggest which part of the table sorter plugin i will need to modify

Wow, okay... In the demo you shared, a minified version of 2.1.2 is being used which would be very difficult to read and modify, so get the [unminified version here](https://cdnjs.cloudflare.com/ajax/libs/jquery.tablesorter/2.1.2/js/jquery.tablesorter.js).

* Open the file in a "good" text editor.
* Go to line 654-655 to find:

    ```js
    .mousedown(function(){
      if (c.cancelSelection) {
    ```

  and change it to

    ```js
    .mousedown(function(e){
      if (c.cancelSelection && e.target.nodeName !== 'SELECT') {
    ```

* Now copy the entire content and paste it into the textarea at https://skalman.github.io/UglifyJS-online/.
* Copy the minified code and save it as `jquery.tablesorter.min.js`.

Please be aware that this change is *very specific* to the demo you shared. It won't work if the column with a select is not disabled or if it contains any other form element (e.g. input or button).

**Comment by gitpuneetsvyas** [comment]
thanks very much for your time and support it's working fine now;your help is very much Appreciated!!!!

**Comment by gitpuneetsvyas** [comment]
Hi Mottie,
this is working fine however when i make a ajax call on the change of select drop down and render the response in div the header class from the other columns disappeared and table sorter UI looks distorted.
Could you please suggest why such behavior of table sorter?

**Comment by gitpuneetsvyas** [comment]
this is the code

`
<body onload="generatePOPDF();">
<div id="rentalStatusResults">	


<form style="margin-top:0;" name="searchForm" id="searchForm" method="post">


<table id="mytable" class="tablesorter">
			
							<thead>
					<tr class="columnHeaders">
						<th>&nbsp;</th>
						<th><jdquote:label key="${Constants.NMBR}"/></th>
						
						<%if(rentalAgreementsSearchBean.isRegion4User()) { %>
						
						
						
				 	
						<th  style="width:25%" align="left" class="rentalStatus" id="byRentalStatus">
						<jdquote:label key="${Constants.STATUS}"/><select disabled="false" align="right"  id="rentalStatus"  onChange="filterRentalAgreementByStatus('filterDataForMYRental','SearchRentalAgreementsServlet');">
								 <c:forEach var="rentalStatus" items="${RentalAgreementsSearchBean.rentalStatusList}">
							 		<option value="${rentalStatus.code}" ${rentalStatus.code==filterRentalStatusId ? 'selected="selected"':""}>${rentalStatus.getDescription()}</option>
							 	</c:forEach>
						</select>
							
													
						</th>
						
						
						
						
			 		
						<% } %>	
						<th><jdquote:label key="${Constants.JDQ_LSE}"/></th>
						<th><jdquote:label key="${Constants.PRODUCTS}"/></th>
						<th><jdquote:label key="${Constants.RATE}"/></th>
						<th><jdquote:label key="${Constants.DAY_OUT}"/></th>
						<th><jdquote:label key="${Constants.DT_DUE_BK}"/></th>
					</tr>
				</thead>
				 
				<%if(searchRentalResultBeans.size() > 0) { %>
					<tfoot>
						<tr id='deleteButton' class="action" >
							<td colspan="6">
						  		<a href='#' class="linkSelect" onclick="javascript:deleteRentals();" ><%=IfcStringLiterals.SEARCH_QUOTE_PAGE_DELETE_SELECTED %></a>
								
							</td>
						</tr>
					</tfoot>
				<% } %>	
					
				
				<tbody>
				
				
				
				<%if(searchRentalResultBeans.size() ==0) { %>
					<tr style="display: none;">
						<td></td>
						<td></td>
						<td></td>
						<td></td>
						<td></td>
						<td></td>
						<td></td>
					</tr>
					<tr class="orderFormSearchResultRow">
						<td colspan="6" align="center"><jdquote:label key="${Constants.NO_CUR_RNTL_AGRMNT_EXSTS}"/></td>
					</tr>
				<%}else{%>	
					<% for(int i=0; i<searchRentalResultBeans.size(); i++) { %>
						<tr class="orderFormSearchResultRow">
									
							<td class="deleteSelected">
								 <input type='checkbox' id='selectedRentals<%=searchRentalResultBeans.get(i).getPurchseOrderId()%>'  name='selectedRentals' onclick="setDirtyFlagForAlert(document.forms[1]);"  value="<%=searchRentalResultBeans.get(i).getPurchseOrderId() %>"> 
							</td>
									
							<td class="orderFormNumber">
					    	<a href="MaintainRentalAgreementServlet?userAction=viewSelectedRentalAgreementFromSearch&orderId=<%=searchRentalResultBeans.get(i).getPurchseOrderId()%>" onClick="javascript:blockBackGround();">
							<jdquote:label key="<%= searchRentalResultBeans.get(i).getRentalNumber()%>"/>
							</td>
							
							
							<td  class="rentalStatus"><%= searchRentalResultBeans.get(i).getStatus()%> </td>	   
							
								
			           		<td class="customerName"><%= searchRentalResultBeans.get(i).getCustomerName()%> </td>
			                <%-- <% String[] product = (String[])searchRentalResultBeans.get(i).getProduct(); %> --%>
							<td title="<%=searchRentalResultBeans.get(i).getProduct()%>" ><%= searchRentalResultBeans.get(i).getProduct()%> </td>
							<td><%= searchRentalResultBeans.get(i).getRate()%> </td>
			                <td  class="dateColumn"><%=searchRentalResultBeans.get(i).getStartDate()%> </td>
			               	<td  class="dateColumn"><%= searchRentalResultBeans.get(i).getEndDate()%> </td>
			   			</tr>
					<% } %>	
				 <%} %> 
				 
				 
				 
				 
			
				 
				 
				 
				 
				</tbody>				
			</table>
		</tr>
</table>
		
		
				
</form>

</div>
</body>	`

**Comment by gitpuneetsvyas** [comment]
Kindly help appreciate your time and help!!!!

**Comment by Mottie** [comment]
If the entire table or any content within the `<thead>` is being replaced by the processed ajax, I'd recommend to first [`destroy`](https://mottie.github.io/tablesorter/docs/#destroy) tablesorter, then initialize it on the new table. If you're only replacing content within the `<tbody>` then trigger an [`update` method](https://mottie.github.io/tablesorter/docs/#update).

You might want to also double check the *rendered* HTML because the HTML shared above has a mysterious trailing `</tr>` that should be removed; unless it really is a nested table being added. Either way, I'd still double check the rendered HTML.

**Comment by Mottie** [comment]
I'm guessing this issue has been resolved, so I'm going to close it. If you continue to have problems, please feel free to continue the discussion in this thread.

**Comment by gitpuneetsvyas** [comment]
Many Many Thanks it is working fine now;

**Comment by gitpuneetsvyas** [comment]
Hi Mottie,
I have found another issue linked to this and related to pager,when i have added ajax for the change of drop down  the pager stopped working i can see records are coming however pager is not showing page 1/1 and page 1/2 which it was showing earlier do you have any pointers? below is the code

`
<TABLE width="100%" border="0" cellpadding=" 0" cellspacing="0">
	<tr>
		<td align="left" valign="top"> 
			<jsp:include page="fragments/header.jsp" flush="false"></jsp:include>
		</td>
	</tr>
</TABLE>
<body onload="generatePOPDF()">


<%-- IE8 Compatibility Fix--%>
<form style="margin-top:0;" name="searchForm" id="searchForm" method="post">	
<table border="0" cellspacing="0" class="layout" width="100%">
<tr>
<input type="hidden" name="dirtyFlagForAlert">
<input type="hidden" name="userAction" value="" />
<input type="hidden" name="pageName" value="currentRentals" />
<input type="hidden" name="pageNumber"/>

				
		
		<td id="leftColumn" nowrap="nowrap">
			<jsp:include page="fragments/leftNavigation.jsp" flush="false"></jsp:include>
		</td>
		
		
		<td id="rightColumn">	
			
				<h1><jdquote:label key="${Constants.MY_CUR_RENTL_AGRMNT}"/></h1>
					<div class="tablesorterCaption">
							<div id="pager" class="tablesorterPager">
									<fieldset><legend><jdquote:label key="${Constants.PAGER}"/></legend>
										<img src="${rentalAgreementsSearchBean.cssPath}/tablesorter/images/first.png" class="first" alt='<jdquote:label key="${Constants.FIRST}"/>' />
										<img src="${rentalAgreementsSearchBean.cssPath}/tablesorter/images/prev.png" class="prev" alt='<jdquote:label key="${Constants.PREVIOUS}"/>' />
										<input type="text" disabled class="pagedisplay"/>
										<img src="${rentalAgreementsSearchBean.cssPath}/tablesorter/images/next.png" class="next" alt='<jdquote:label key="${Constants.NEXT}"/>' />
										<img src="${rentalAgreementsSearchBean.cssPath}/tablesorter/images/last.png" class="last" alt=<jdquote:label key="${Constants.LAST}"/> />
										<select class="pagesize">
											<option value="10" selected="selected">10</option>
											<option value="20">20</option>
											<option value="30">30</option>
											<option value="40">40</option>
										</select>
									</fieldset>
								</div>
					</div>
					<div id="rentalStatusResults">	
			<jsp:include page="MyRentalStatus.jsp" flush="false" />	
			 </div>
		</tr>
</table>
		
		
				
</form>	


`

Content of Included JSP

`
<table id="mytable" class="tablesorter">
			
							<thead>
					<tr class="columnHeaders">
						<th>&nbsp;</th>
						
						<th><jdquote:label key="${Constants.NMBR}"/></th>
						
						<%if(rentalAgreementsSearchBean.isRegion4User()) { %>
												
						
	
						<th  style="width:25%" align="left" class="rentalStatus" id="byRentalStatus">
						<jdquote:label key="${Constants.STATUS}"/><select disabled="false" align="right"  id="rentalStatus"  onChange="filterRentalAgreementByStatus('filterDataForMYRental','SearchRentalAgreementsServlet');">
								 <c:forEach var="rentalStatus" items="${RentalAgreementsSearchBean.rentalStatusList}">
							 		<option value="${rentalStatus.code}" ${rentalStatus.code==filterRentalStatusId ? 'selected="selected"':""}>${rentalStatus.getDescription()}</option>
							 	</c:forEach>
						</select>
							
													
						</th>
						
						<% } %>	
						<th><jdquote:label key="${Constants.JDQ_LSE}"/></th>
						<th><jdquote:label key="${Constants.PRODUCTS}"/></th>
						<th><jdquote:label key="${Constants.RATE}"/></th>
						<th><jdquote:label key="${Constants.DAY_OUT}"/></th>
						<th><jdquote:label key="${Constants.DT_DUE_BK}"/></th>
					</tr>
				</thead>
				
				<%if(rentalAgreementsSearchBean.getNumber().length > 0) { %>
					<tfoot>
						<tr id='deleteButton' class="action" >
							<td colspan="6">
						  		<a href='#' class="linkSelect" onclick="javascript:deleteRentals();" ><%=IfcStringLiterals.SEARCH_QUOTE_PAGE_DELETE_SELECTED %></a>
								
							</td>
						</tr>
					</tfoot>
				<% } %>	
					
				
				<tbody>
				
				
				
				<%if(rentalAgreementsSearchBean.getNumber().length ==0) { %>
					<tr style="display: none;">
						<td></td>
						<td></td>
						<td></td>
						<td></td>
						<td></td>
						<td></td>
						<td></td>
					</tr>
					<tr class="orderFormSearchResultRow">
						<td colspan="6" align="center"><jdquote:label key="${Constants.NO_CUR_RNTL_AGRMNT_EXSTS}"/></td>
					</tr>
				<%}else{%>	
					<% for(int i=0; i<rentalAgreementsSearchBean.getNumber().length; i++) { %>
					
						<tr class="orderFormSearchResultRow">
									
							<td class="deleteSelected">
								 <input type='checkbox' id='selectedRentals<%=rentalAgreementsSearchBean.getPurchseOrderIds()[i]%>'  name='selectedRentals' onclick="setDirtyFlagForAlert(document.forms[1]);"  value="<%=rentalAgreementsSearchBean.getPurchseOrderIds()[i]%>"> 
							</td>
									
							<td class="orderFormNumber">
					    	<a href="MaintainRentalAgreementServlet?userAction=viewSelectedRentalAgreementFromSearch&orderId=<%=rentalAgreementsSearchBean.getPurchseOrderIds()[i] %>" onClick="javascript:blockBackGround();">
							
							<jdquote:label key="<%= rentalAgreementsSearchBean.getNumber()[i]%>"/>
					<%-- 	<c:out value="<%= rentalAgreementsSearchBean.get(i).getNumber()%>"/> --%>
							</td>
							
							 <c:choose>
							  <c:when test="${region4User}">
							<td class="rentalStatus"><%= rentalAgreementsSearchBean.getStatus()[i]%> </td>	   
							</c:when>
							  <c:otherwise>
					     
					      </c:otherwise>
				          </c:choose>
								
			           		<td class="customerName"><%= rentalAgreementsSearchBean.getCustomer()[i]%> </td>
			           		<% String[] product = (String[])rentalAgreementsSearchBean.getProducts().get(i); %>			         		
							<td title="<%=product[1]%>" ><%= product[0]%> </td>
							<td><%= rentalAgreementsSearchBean.getRates()[i]%> </td>
			                <td  class="dateColumn"><%= rentalAgreementsSearchBean.getStartDates()[i]%> </td>
			               	<td  class="dateColumn"><%= rentalAgreementsSearchBean.getEndDates()[i]%> </td>
			   			</tr>
					<% } %>	
				 <%} %> 
				 </tbody>				
			</table>

`

**Comment by gitpuneetsvyas** [comment]
any pointers?

**Comment by thezoggy** [comment]
This is not the proper support channel to get help with how to write code, this is for bug support with the current released version of the code. You've already noted that you are using an outdated legacy version, we can not provide support for this. tablesorter v2.06 was released on Jun 22, 2011 - 6 years, 8 months, 26 days ago!

**Comment by Mottie** [comment]
I need more info:

* In the minified [code](https://github.com/Mottie/tablesorter/issues/1510#issuecomment-365611338) shared above, I don't see anything dealing with ajax. Please share that code.
* Which `select`? The page size or the rental status? Is there a page number select not shown?
* The default [`output` option](https://mottie.github.io/tablesorter/docs/#pager-output) is set to `'{startRow} to {endRow} of {totalRows} rows'`, it was changed from `'{page}/{totalPages}'` in v2.0.22 (I forgot to update the docs). And, I don't see the `output` option being changed. What version of the pager are you using?
* I see the targeted [`cssPageDisplay` input](https://mottie.github.io/tablesorter/docs/#pager-csspagedisplay) (using a span would be better) in the rendered HTML, but not in the JSP; could this be the issue?

**Comment by gitpuneetsvyas** [comment]
thanks mottie for Pointers it is working fine now table.trigger()'update') after ajax response was causing  the problem.thanks for your time and support
